cmake/Modules/FindXIOS.cmake

```
@@ -42,15 +50,29 @@ SET(XIOSLIB
   "$ENV{XIOS_ROOT}/lib"
   INTERNAL)
 
+# Prefer shared (.so/.dll/.dylib) over static lib (.a/.lib) if both are present
+IF(APPLE)
+  SET(CMAKE_FIND_LIBRARY_SUFFIXES .dylib .a)
+ELSEIF(WIN32)
+  SET(CMAKE_FIND_LIBRARY_SUFFIXES .dll .lib)
+ELSE()
+  SET(CMAKE_FIND_LIBRARY_SUFFIXES .so .a)
+ENDIF()
+
 FIND_LIBRARY(XIOS_LIBRARY xios HINTS ${XIOSLIB})
 
 IF (XIOS_INCLUDE_DIR AND XIOS_LIBRARY)
-  UNSET(XIOS_FAILMSG)
-  SET(XIOSLIB_FOUND TRUE)
-  SET(XIOS_INCLUDE_DIR ${XIOS_INCLUDE_DIR})
-  SET(XIOS_LIBRARIES "${XIOS_LIBRARY}")
+  GET_FILENAME_COMPONENT(XIOS_LIBRARY_EXT ${XIOS_LIBRARY} EXT)
+  IF(XIOS_LIBRARY_EXT MATCHES "so|dll|dylib")
+    UNSET(XIOS_FAILMSG)
+    SET(XIOSLIB_FOUND TRUE)
+    SET(XIOS_INCLUDE_DIR ${XIOS_INCLUDE_DIR})
+    SET(XIOS_LIBRARIES "${XIOS_LIBRARY}")
+  ELSE()
+    SET(XIOS_FAILMSG "XIOS only found as a static library. Please rebuild XIOS as a shared library (.so/.dll/.dylib) if you want to use it.")
+  ENDIF()
```

This part automatically checks and enforces what's documented about XIOS being a shared lib in the Elmer documentation here: https://github.com/ElmerCSC/elmerfem/blob/2492b9cd3a39e05f1c1088aadafd0a2e3cc0aac4/elmerice/Solvers/Documentation/XIOSOutputSolver.md?plain=1#L30-L39

Motivation: Let this fail early if XIOS has not be built correctly. Otherwise the user will waste a lot of time trying to build elmer and eventually fail.